### PR TITLE
Add sub-modules support

### PR DESCRIPTION
**Pull request by jaredfholgate**
Add the capability to generate documentation, check documentation, lint and validate for sub modules stored in the standard `modules` folder.

All of these checks are skipped if the `modules` folder is missing or does not contain any sub folders. Therefore this should be a non-breaking change and only help modules owners leveraging sub modules.

There is a sister PR for the template ignore file here: https://github.com/Azure/terraform-azurerm-avm-template/pull/119

### CODE REVIEW

**Comment by jaredfholgate** [comment]
@matt-FFFFFF or @lonegunmanb What would be the easiest way for me to test these changes locally?

**Comment by matt-FFFFFF** [comment]
Put them in canary dir first, then set the AVM_ENVIRONMENT variable do canary before running the make command

**Comment by matt-FFFFFF** [comment]
Sorry. AVMSCRIPT_VERSION=canary

**Comment by matt-FFFFFF** [comment]
Or just run it from the directory of your module

**Comment by jaredfholgate** [comment]
I found an approach that works for me. 

- Use the devcontainer in vscode
- Mount my user profile in the dev container with `"source=${localEnv:HOME}${localEnv:USERPROFILE},target=/host-home-folder,type=bind,consistency=cached"`
- cd to the repo I want to test in the mount
- Run the shell scripts using `bash /workspaces/tfmod-scaffold/avm_scripts/autofix.sh`

**Comment by jaredfholgate** [comment]
@matt-FFFFFF and @lonegunmanb I have successfully tested the updated scripts now.

File: avm_scripts/terraform-validate.sh [hunk 1]
```diff
@@ -24,4 +24,24 @@ for d in $examples; do
      exit 1
   fi
 done
+
+echo "==> Checking module terraform code are validate..."
+if [ ! -d modules ]; then
+  echo "==> Warning - no modules directory found"
+else
+  examples=$(find ./modules -maxdepth 1 -mindepth 1 -type d)
```
Comment (matt-FFFFFF) [inline avm_scripts/terraform-validate.sh:32]: nit but can we rename this variable? and the singular form below?

File: avm_scripts_canary/terraform-validate.sh [hunk 2]
```diff
@@ -24,4 +24,24 @@ for d in $examples; do
      exit 1
   fi
 done
+
+echo "==> Checking module terraform code are validate..."
+if [ ! -d modules ]; then
+  echo "==> Warning - no modules directory found"
+else
+  examples=$(find ./modules -maxdepth 1 -mindepth 1 -type d)
```
Comment (matt-FFFFFF) [inline avm_scripts_canary/terraform-validate.sh:32]: nit - same please can we name to modules/module?

**Comment by jaredfholgate** [comment]
To be clear on the tests run for this. All five scripts were run with:

- `modules` folder with a module inside it.
- `modules` folder with no modules inside it.
- No `modules` folder.

All behaved as expected in a backwards compatible manner. If there are no sub modules this change has no impact. If there are sub module it will be able to fix, document, validate and lint them.

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**